code/modules/surgery/organs/organ_internal.dm

```diff
@@ -248,12 +256,246 @@
+				H.adjustOxyLoss(-5)
+			gas_breathed = breath_gases["o2"][MOLES]
+			H.clear_alert("oxy")
+
+	//Exhale
+	breath_gases["o2"][MOLES] -= gas_breathed
+	breath_gases["co2"][MOLES] += gas_breathed
+	gas_breathed = 0
+
+
+	//-- CO2 --//
+
+	//CO2 does not affect failed_last_breath. So if there was enough oxygen in the air but too much co2, this will hurt you, but only once per 4 ticks, instead of once per tick.
+	if(breathlevels["safe_co2_max"])
+		if(CO2_pp > breathlevels["safe_co2_max"] && !(NOBREATH in specflags))
+			if(!H.co2overloadtime) // If it's the first breath with too much CO2 in it, lets start a counter, then have them pass out after 12s or so.
+				H.co2overloadtime = world.time
+			else if(world.time - H.co2overloadtime > 120)
+				H.Paralyse(3)
+				H.adjustOxyLoss(3) // Lets hurt em a little, let them know we mean business
+				if(world.time - H.co2overloadtime > 300) // They've been in here 30s now, lets start to kill them for their own good!
+					H.adjustOxyLoss(8)
+				H.throw_alert("too_much_co2", /obj/screen/alert/too_much_co2)
+			if(prob(20)) // Lets give them some chance to know somethings not right though I guess.
+				H.emote("cough")
+
+		else
+			H.co2overloadtime = 0
+			H.clear_alert("too_much_co2")
+
+	//Too little CO2!
+	if(breathlevels["safe_co2_min"])
+		if(CO2_pp < breathlevels["safe_co2_min"])
+			gas_breathed = handle_too_little_breath(H,CO2_pp, breathlevels["safe_co2_min"],breath_gases["co2"][MOLES])
+			H.throw_alert("not_enough_co2", /obj/screen/alert/not_enough_co2)
+		else
+			H.failed_last_breath = 0
+			H.adjustOxyLoss(-5)
+			gas_breathed = breath_gases["co2"][MOLES]
+			H.clear_alert("not_enough_co2")
+
+	//Exhale
+	breath_gases["co2"][MOLES] -= gas_breathed
+	breath_gases["o2"][MOLES] += gas_breathed
+	gas_breathed = 0
+
+
+	//-- TOX --//
+
+	//Too much toxins!
+	if(breathlevels["safe_toxins_max"])
+		if(Toxins_pp > breathlevels["safe_toxins_max"] && !(NOBREATH in specflags))
+			var/ratio = (breath_gases["plasma"][MOLES]/breathlevels["safe_toxins_max"]) * 10
+			if(H.reagents)
+				H.reagents.add_reagent("plasma", Clamp(ratio, tox_breath_dam_min, tox_breath_dam_max))
+			H.throw_alert("tox_in_air", /obj/screen/alert/tox_in_air)
+		else
+			H.clear_alert("tox_in_air")
+
+
+	//Too little toxins!
+	if(breathlevels["safe_toxins_min"])
+		if(Toxins_pp < breathlevels["safe_toxins_min"] && !(NOBREATH in specflags))
+			gas_breathed = handle_too_little_breath(H,Toxins_pp, breathlevels["safe_toxins_min"], breath_gases["plasma"][MOLES])
+			H.throw_alert("not_enough_tox", /obj/screen/alert/not_enough_tox)
+		else
+			H.failed_last_breath = 0
+			H.adjustOxyLoss(-5)
+			gas_breathed = breath_gases["plasma"][MOLES]
+			H.clear_alert("not_enough_tox")
+
+	//Exhale
+	breath_gases["plasma"][MOLES] -= gas_breathed
+	breath_gases["co2"][MOLES] += gas_breathed
+	gas_breathed = 0
+
+
+	//-- TRACES --//
+
+	if(breath && !(NOBREATH in specflags))	// If there's some other shit in the air lets deal with it here.
```

Comment:
Ok, tested it. If the species has nobreath, they basically skip breathe(), and don't call check_breath() which is where the lung code is.
So they don't breathe if you transplant lungs into them.